Framework/Algorithms/src/SampleCorrections/MCAbsorptionStrategy.cpp

```diff
@@ -74,6 +75,9 @@ MCAbsorptionStrategy::calculate(Kernel::PseudoRandomNumberGenerator &rng,
       }
     } while (true);
   }
+
+  debugString += m_scatterVol.generateScatterPointStats();
```

Comment:
Just a quick question. Is generating and logging the string impactful on performance at all? As it's only set to debug we are mostly doing extra work that's thrown away. It's possible to test the [Logger level](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/inc/MantidKernel/Logger.h#L122) so we could avoid creating the string if the log level is not debug?

Reply:
I don't think it's having a big impact because it's just incrementing 6-7 counters. I can add this in though because it sounds like generally good practice and I guess the logging could become heavier in future. Will need to pass the parent algorithm's logger into the strategy I think (instead of the debug string)

Reply:
Sounds good. Thanks.